python/lsst/ap/association/diaPipe.py

```diff
@@ -97,17 +97,31 @@ class DiaPipelineConnections(
     )
     associatedDiaSources = connTypes.Output(
         doc="Optional output storing the DiaSource catalog after matching, "
-            "calibration, and standardization for insertation into the Apdb.",
+            "calibration, and standardization for insertion into the Apdb.",
         name="{fakesType}{coaddName}Diff_assocDiaSrc",
         storageClass="DataFrame",
         dimensions=("instrument", "visit", "detector"),
     )
+    diaForcedSources = connTypes.Output(
+        doc="Optional output storing the forced sources computed at the diaObject positions.",
+        name="{fakesType}{coaddName}Diff_diaForcedSource",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    diaObjects = connTypes.Output(
```

Comment:
Please investigate how to make the `validityStart` column behave in a more useful way. Ideally it would be updated to the given exposure's timestamp before this table is persisted. Right now that doesn't seem to be happening, and instead `validityStart` times are only pulled in via the APDB DiaObject history. This makes it next to impossible to construct an APDB-equivalent DiaObject Table from this.

Reply:
Can you give me a minimum working example on how to demonstrate this difference? I think that digging into how the diaObject table is created is well outside the scope of this ticket, but if it's an easy change to which thing I'm persisting, I can give it a try.

Reply:
Here's a MWE showing the difference:

```#!/usr/bin/env python
# script from Meredith to compare validityStart values between APDB and the new parque output.
import os
import sys

import pandas as pd
from IPython.display import display
import lsst.daf.butler as dafButler
sys.path.append('/sdf/data/rubin/u/mrawls/scipipe/analysis_ap/python/lsst/analysis/ap/')
import apdbUtils as pla

repo = '/sdf/data/rubin/u/parejko/cosmos-DM-36199/repo'
collections = 'ap_verify-output'
butler = dafButler.Butler(repo, collections=collections)
instrument = 'HSC'
skymap = 'hsc_rings_v1'
dataIds = [{'visit': 59150, 'detector': 50, 'instrument': 'HSC'}, {'visit': 59160, 'detector': 51, 'instrument': 'HSC'}]

# Load APDB DiaObject table
objTableApdb = pla.loadAllApdbObjects(dbName=os.path.join(repo, '../association.db'), allCol=True)
objTableApdb.set_index('diaObjectId', inplace=True)

# Load parquet DiaObject tables and concatenate them, which is necessary to make desired "per-run" plots
def loadTable(dataId, dataset):
    oneTable = butler.get(dataset, dataId=dataId)
    return oneTable

objFrames = [loadTable(dataId, 'fakes_deepDiff_diaObject') for dataId in dataIds]
objTableParquet = pd.concat(objFrames)
objTableParquet.drop(columns='diaObjectId', inplace=True)

# Compare the two tables' validityStart and validityEnd columns
display(objTableApdb.sort_index().sort_index(axis=1)[['nDiaSources', 'validityStart', 'validityEnd']])
display(objTableParquet.sort_index().sort_index(axis=1)[['nDiaSources', 'validityStart', 'validityEnd']])
```

Reply:
Having dug into this a bit more, all of the validity information is computed inside dax_apdb and done in the database itself. We don't have a facility right now to move that information back into the diaObject structure inside `diaPipe.py`, and I think figuring out how to do that is out of scope for this ticket. I'd advocate merging this as-is, so that we at least have some output written, and file another ticket to (optionally?) have the updated values of filled in inside the dax_apdb code (`_storeDiaObjects`, but those seem quite different in `apdbSql.py` vs. `abdpCassandra.py`, and there's no abstractmethod in the base class).